src/message/messageActionSheet.js

```diff
@@ -120,14 +126,28 @@ const unstarMessage = async ({ auth, message }) => {
 unstarMessage.title = 'Unstar message';
 unstarMessage.errorMessage = 'Failed to unstar message';
 
-const shareMessage = ({ message }) => {
+const shareMessage = async ({ auth, message }) => {
+  if (message.isOutbox === true) {
+    throw new Error('Message has not been sent.');
+  }
+  const rawMessage = await getRawMessageContent(auth, message);
   Share.share({
-    message: message.content.replace(/<(?:.|\n)*?>/gm, ''),
+    message: `${rawMessage}\n\n— ${getLinkToMessage(auth.realm, message)}`,
   });
 };
 shareMessage.title = 'Share';
 shareMessage.errorMessage = 'Failed to share message';
 
+const copyMessageLink = ({ _, auth, message }) => {
+  if (message.isOutbox === true) {
+    throw new Error('Message has not been sent.');
+  }
+  Clipboard.setString(getLinkToMessage(auth.realm, message));
+  showToast(_('Copied link to message'));
+};
+copyMessageLink.title = 'Copy message link';
```

Comment:
I looked at this and thought "gosh, that sounds ambiguous -- what about when the message *contains* a link? won't it sound like this means to copy that link?"

Then I found there'd been discussion of exactly that on the previous PR: https://github.com/zulip/zulip-mobile/pull/3412#issuecomment-502324323 :slightly_smiling_face: and I guess my conclusion was:
> I think it's still a little ambiguous what "message link" means [...] Is it "Copy a link to the message", or "Copy the link in the message"? But it's good enough I'll be happy to merge it, and we can wordsmith further later.

So that seems reasonable -- I'll stand by that. If you have some ideas for that further wordsmithing, though, I'm interested!

Reply:
Considering that the success string just above is "Copied link to message", I'd say present-tensing that as "Copy link to message" is probably for the best.